datafusion/expr/src/logical_plan/plan.rs

```diff
@@ -396,8 +396,10 @@ impl LogicalPlan {
         }
 
         let recurse = match self {
-            LogicalPlan::Projection(Projection { input, .. }) => input.accept(visitor)?,
-            LogicalPlan::Filter(Filter { input, .. }) => input.accept(visitor)?,
+            LogicalPlan::Projection(Projection { .. }) => {
+                self.visit_all_inputs(visitor)?
+            }
+            LogicalPlan::Filter(Filter { .. }) => self.visit_all_inputs(visitor)?,
```

Comment:
This is the main change. We now visit all inputs, including subqueries. I have only implemented this for `Projection` and `Filter` in this PR.